src/server/authserver/Server/AuthSession.cpp

```diff
@@ -415,32 +406,46 @@ void AuthSession::LogonChallengeCallback(PreparedQueryResult result)
     TC_LOG_DEBUG("network", "database authentication values: v='%s' s='%s'", databaseV.c_str(), databaseS.c_str());
 
     // multiply with 2 since bytes are stored as hexstring
-    if (databaseV.size() != size_t(BufferSizes::SRP_6_V) * 2 || databaseS.size() != size_t(BufferSizes::SRP_6_S) * 2)
-        SetVSFields(rI);
+    if (databaseV.empty() || databaseS.empty())
+    {
+
+#ifndef _MSC_VER
```

Comment:
Don't use _MSC_VER directly anywhere, we have `TRINITY_COMPILER` defined in CompilerDefs.h

Reply:
Actually, dont mark the function are deprecated using the attribute then suppress it all over the place... people that use them in their custom code probably don't care about warnings so all you are doing is making our code ugly. Put it in function name `MakeRegistrationDataFromHash_DEPRECATED`